### ISSUE TITLE
brew cask alfred transition

### ISSUE THREAD

**Issue by rolandwalker** [caskroom/homebrew-cask#9405]
Support for manipulating Alfred scopes was removed in #8840 due to direct support from Alfred (https://github.com/caskroom/homebrew-cask/issues/8052#issuecomment-69560606).

But a disappeared feature is disorienting (#8840).  There's no doc for the user to explain what is missing.

`brew cask alfred` should be brought back, as a feature-free stub that emits a message about why it is no longer needed.  After a few months, the stub should be removed.

**Comment by Pherocity** [comment]
Or at least make a bloody note of it on the readme or usage page.  Seriously, this wasn't thought out at all.  I have to dig through issues to find out what I need to manually do?

**Comment by rolandwalker** [comment]
Sorry for the inconvenience!  I feel strongly about that, as convenience is really the sole purpose of this software.  We'll make some changes within a few days.

**Comment by phinze** [comment]
> Or at least make a bloody note of it on the readme or usage page. Seriously, this wasn't thought out at all.

Yowza - such anger! I'll never understand the desire to inject vitriol into an all-volunteer open source project.

Oh well  `¯\_(ツ)_/¯` - this will be fixed with a stub command and a friendly note in the next release! :+1:

**Comment by vitorgalvao** [comment]
I was interrupted when writing this comment and the issue has since been addressed, but since I was so long into it, might as well finish and post it anyway.

Apologies on causing the confusion, @Pherocity. I didn’t anticipate so many confused users when making the change, and even though I agree with making a note (as stated in this issue) I hadn’t had the chance to actually work on it. For some context, part of my eagerness to remove the feature was it wasn’t working correctly anyway, so getting rid of it altogether was very appealing.

I’m also one to be very blunt<sup>1</sup>, and that makes many of my comments unreasonably<sup>2</sup> harsh. That leads to some of my replies taking a confrontational tone, like if I had lost my temper, when that is not how I in fact feel. That (and clarity of expressed ideas) is why I edit my comments before posting and tweak them right after. A lot.

Even so, the initiator of this project, @phinze, who was the one to first add me as a collaborator, during this project’s infancy, to whom I greatly thank the gesture and have a lot of respect for, has unfortunately and unfairly been on the receiving end of more than one of my unedited posts, where I just ramble on with a shameful<sup>3</sup> attack tone.

This is not only a further apology for those cases, but also to say that whenever I see a comment that looks harsh, I always try to interpret it first as if the commenter said it with the best of tones, as it helps to clear ambiguities of this kind for the better. Yours, however, briefly reminded me of [this issue](https://github.com/steveklabnik/emoji/issues/23) (on a smaller scale) that was on HackerNews a while back.

I ask that you reconsider the entitlement present in your comment, particularly in the last sentence, for a project that is (hopefully) making your life easier otherwise, is very young and unready for production in many areas (although already very popular and widely used), and kept up by a small team of regular maintainers (and thankfully some regular contributors), all volunteering their time, making a conscious decision to spend it improving a community project instead of doing something else.

In the time it took any of us to write our comments, a submission could’ve been made to fix it instead, or at the very least make the mentioned note in the documentation. However, I still believe this point was worth making.

Kudos to @phinze for doing it so quickly and still keeping composure when answering these comments, and to @rolandwalker for similarly always keeping his calm. Thank you both for bringing this project where it is. It’s a pleasure working with you.

---

<sup>1</sup> I do write a lot (probably too much), but I do not like to skirt around opinions.
<sup>2</sup> That being the keyword, there.
<sup>3</sup> Reasons are irrelevant. They don’t excuse the behaviour.

**Comment by Pherocity** [comment]
You are correct, I was angry because someone moved my cheese.  I then wasted an hour dedicated to something else because I couldn't figure out where my cheese was, assuming I did something wrong.  My day is slightly bent out of shape, and now I'll be working over my lunch.  I get frustrated when developers haphazardly break tools used for automated deployments.  Because if I can't trust your tool, then I need to find another that I can trust.  I've sung the praises of the cask tool, but if there is no oversight to breaking changes with little more than a commit, I'm making bad recommendations. This in no way excuses my behaviour, and was entirely uncalled for.  It is unacceptable, I am very sorry for any offence.
